ros_buildfarm/status_page.py

```diff
@@ -39,6 +39,111 @@
+        repos_info[repo_name]['repos_blocked_by'] = {}
+        repos_info[repo_name]['maintainers'] = defaultdict(dict)
+
+    for repo_name in prev_released_repos:
+        repos_info[repo_name]['url'] = _repo_url(prev_dist, repo_name)
+
+    # has a source entry? Lets call that 'released' with a 'version' to reuse
+    # logic for the blocked releases page
+    for repo_name in source_entry_repos:
+        repos_info[repo_name]['released'] = True
+        repos_info[repo_name]['version'] = 'yes'
+
+    # Determine which repos directly block the missing ones
+    for repo_name in missing_repos:
+        package_dependencies = set()
+        for pkg in _released_packages(prev_dist, (repo_name,)):
+            package_dependencies.update(_package_dependencies(prev_walker, pkg))
+
+        for dep in package_dependencies:
+            if dep in missing_packages:
+                blocking_repo = prev_dist.release_packages[dep].repository_name
+                if blocking_repo == repo_name:
+                    # ignore packages in the same repo
+                    continue
+                url = _repo_url(prev_dist, repo_name)
+                repos_info[repo_name]['repos_blocked_by'][blocking_repo] = url
+                new_maintainers = dict(_maintainers(prev_dist, dep))
+                repos_info[repo_name]['maintainers'][blocking_repo].update(new_maintainers)
+
+                # Mark blocking relationship in other direction
+                repos_info[blocking_repo]['repos_blocking'].add(repo_name)
+
+    # Compute which repos block another recursively
+    for repo_name in repos_info.keys():
+        checked_repos = set()
+        repos_to_check = set([repo_name])
+        while repos_to_check:
+            next_repo = repos_to_check.pop()
+            new_repos_blocking = repos_info[next_repo]['repos_blocking']
+            repos_info[repo_name]['recursive_repos_blocking'].update(new_repos_blocking)
+            checked_repos.add(next_repo)
+            repos_to_check.update(repos_info[next_repo]['repos_blocking'].difference(checked_repos))
+
+    # Expand a template for the webpage
+    repos_data = [_format_repo_table_row(name, data) for name, data in sorted(repos_info.items())]
+    template_name = 'status/blocked_source_entries_page.html.em'
+    data = {
+        'title': 'ROS %s - blocked source entries' % rosdistro_name,
+        'start_time': time.strftime('%Y-%m-%d %H:%M:%S %z', start_time),
+        'resource_hashes': get_resource_hashes(),
+        'rosdistro_name': rosdistro_name.capitalize(),
+        'repos_data': repos_data,
+    }
+    output_filename = os.path.join(
+        output_dir, 'blocked_source_entries_%s.html' % rosdistro_name)
+    print("Generating blocked source entries page '%s':" % output_filename)
+    with open(output_filename, 'w') as h:
+        h.write(expand_template(template_name, data))
+    additional_resources(output_dir, copy_resources=copy_resources)
```

Comment:
How about splitting this function in a similar way as the existing one for blocked releases: `_get_blocked_releases_info` focuses on collecting the data and `build_blocked_releases_page` which calls the former takes care of the template part and the formatting?

Reply:
+1 to smaller functions. Refactored in 334837e